service/src/main/scala/za/co/absa/loginsvc/rest/service/JWTService.scala

```diff
@@ -21,16 +21,16 @@ import io.jsonwebtoken.{Jwts, SignatureAlgorithm}
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
 import za.co.absa.loginsvc.model.User
-import za.co.absa.loginsvc.rest.config.JwtConfig
-
+import za.co.absa.loginsvc.rest.config.ConfigProvider
 import java.security.{KeyPair, PublicKey}
 import java.time.Instant
 import java.time.temporal.ChronoUnit
 import java.util.Date
 
 @Service
-class JWTService @Autowired()(jwtConfig: JwtConfig) {
+class JWTService @Autowired()(configProvider: ConfigProvider) {
```

Comment:
This feels a bit weird, because it is only the JTW config this service should have access to (analogically elsewhere).

I think I understand why you did it this way - ConfigProvider as a `@Component` can be autowired while the individual configs cannot. However, perhaps there is a middle ground - if the type was `JWTConfigProvider` (a trait that regular `ConfigProvider` implements and only has jwt-config-related method) that would hide the other config methods here -- and for the future -- it would pose no assumptions if the config @component should be one or more.

You do not need to follow this to the point, I am trying to explain what seemed a bit off with some background.